crates/notion-core/src/catalog.rs

```diff
@@ -278,51 +280,7 @@ impl RegistryFetchError {
 
 impl Resolve<NodeDistro> for NodeCollection {
     fn resolve_public(&self, matching: &VersionReq) -> Fallible<NodeDistro> {
-        let index: Index = match read_cached_opt().unknown()? {
-            Some(serial) => serial,
-            None => {
-                let spinner = progress_spinner(&format!(
-                    "Fetching public registry: {}",
-                    PUBLIC_NODE_VERSION_INDEX
-                ));
-                let mut response: reqwest::Response = reqwest::get(PUBLIC_NODE_VERSION_INDEX)
-                    .with_context(RegistryFetchError::from_error)?;
-                let response_text: String = response.text().unknown()?;
-                let cached: NamedTempFile = NamedTempFile::new().unknown()?;
-
-                // Block to borrow cached for cached_file.
-                {
-                    let mut cached_file: &File = cached.as_file();
-                    cached_file.write(response_text.as_bytes()).unknown()?;
-                }
-
-                cached.persist(path::node_index_file()?).unknown()?;
-
-                let expiry: NamedTempFile = NamedTempFile::new().unknown()?;
-
-                // Block to borrow expiry for expiry_file.
-                {
-                    let mut expiry_file: &File = expiry.as_file();
-
-                    if let Some(expires_header) = response.headers().get::<Expires>() {
-                        write!(expiry_file, "{}", expires_header).unknown()?;
-                    } else {
-                        let expiry_date =
-                            SystemTime::now() + Duration::from_secs(max_age(&response).into());
-
-                        write!(expiry_file, "{}", HttpDate::from(expiry_date)).unknown()?;
-                    }
-                }
-
-                expiry.persist(path::node_index_expiry_file()?).unknown()?;
-
-                let serial: serial::index::Index =
-                    serde_json::de::from_str(&response_text).unknown()?;
-
-                spinner.finish_and_clear();
-                serial
-            }
-        }.into_index()?;
+        let index: Index = resolve_node_versions().unwrap().into_index()?;
```

Comment:
You should use a question mark here instead of `unwrap()`, so the error from `resolve_node_versions` is propagated correctly

Reply:
👍 good to learning, updated the diff